src/js/modules/setup/controllers/change-password/change-password.ctrl.js

```diff
@@ -0,0 +1,305 @@
+
+            $scope.working = true;
+
+            if ($scope.form.ERS.length > 0) {
+                try {
+                    recoverySecret = recoverySecret.trim();
+                    var encryptedRecoverySecretMnemonic = $scope.form.ERS
+                        .trim()
+                        .replace(new RegExp("\r\n", 'g'), " ")
+                        .replace(new RegExp("\n", 'g'), " ")
+                        .replace(/\s\s+/g, " ")
+                        .replace("[ \t]+$", "");
+                    var encryptedRecoverySecret = null;
+
+                    if ($scope.walletVersions.indexOf(walletVersion) < 0) {
+                        $scope.working = false;
+                        return dialogService.alert(
+                            $translate.instant("INVALID_WALLET_VERSION"),
+                            $translate.instant("MSG_INVALID_WALLET_VERSION")
+                        ).result;
+                    }
+
+                    if (recoverySecret.length != 64) {
+                        $scope.working = false;
+                        return dialogService.alert(
+                            $translate.instant("RECOVERY_ERROR"),
+                            $translate.instant("MSG_CORRUPTED_SECRET")
+                        ).result;
+                    }
+
+                    if ($scope.form.ERS.split(' ').length != 60) {
+                        $scope.working = false;
+                        return dialogService.alert(
+                            $translate.instant("RECOVERY_ERROR"),
+                            $translate.instant("MSG_WRONG_ERS_WORD_LENGTH")
+                        ).result;
+                    }
+
+                    // Try decrypting
+                    try {
+                        if (walletVersion === 'v3') {
+                            encryptedRecoverySecret = blocktrailSDK.EncryptionMnemonic.decode(encryptedRecoverySecretMnemonic);
+                            secret = blocktrailSDK.Encryption.decrypt(encryptedRecoverySecret, new blocktrailSDK.Buffer(recoverySecret, 'hex'));
+
+                        } else {
+                            // TODO: BIP39 not found :/
+                            encryptedRecoverySecret = blocktrailSDK.convert(blocktrailSDK.bip39.mnemonicToEntropy(encryptedRecoverySecretMnemonic), 'hex', 'base64');
+                            secret = CryptoJS.AES.decrypt(encryptedRecoverySecret, recoverySecret).toString(CryptoJS.enc.Utf8);
+                        }
+                    } catch (e) {
+                        $log.error(e, e.message);
+                        $scope.working = false;
+                        return dialogService.alert(
+                            $translate.instant("RECOVERY_ERROR"),
+                            $translate.instant("MSG_PASSWORD_NOT_CHANGED")
+                        ).result;
+                    }
+
+                    watchListener();
+                    console.log(secret);
```

Comment:
strip `console.log` or release with `$log.debug`

also, why do we unsub the `watchListener` ? and maybe better to call that `watchListener` something like `unsubWatchERS`? @romaroid